builtin/game/item.lua

```diff
@@ -454,12 +454,16 @@ function core.item_drop(itemstack, dropper, pos)
 end
 
 function core.do_item_eat(hp_change, replace_with_item, itemstack, user, pointed_thing)
+	-- Don't consume the item if the player is already at full health
+	if user:get_hp() >= core.PLAYER_MAX_HP_DEFAULT then return end
```

Comment:
![](https://i.imgur.com/RUdPyQP.jpg)

Reply:
Nice meme, but it does not suggest what to use instead.
```Lua
if user:get_hp() >= user:get_properties().hp_max then
	return
end
```

Reply:
```lua
	if user:get_hp() >= user:get_properties().hp_max then
		return itemstack
	end
```
https://github.com/minetest/minetest/blob/a978278bfba743ece5ec149aa366067aa47b61ff/doc/lua_api.txt#L4552-L4554